packages/auth/src/auth-options.ts

```diff
@@ -0,0 +1,72 @@
+import { prisma } from "@agreeto/db";
+import { PrismaAdapter } from "@next-auth/prisma-adapter";
+import type { NextAuthOptions } from "next-auth";
+import GoogleProvider from "next-auth/providers/google";
+import AzureAdProvider from "next-auth/providers/azure-ad";
+import { refreshAccessToken } from "./refresh-access-token";
+
+export type GetAuthOptionsParams = {
+  secret: string;
+  googleClientId: string;
+  googleClientSecret: string;
+  azureAdClientId: string;
+  azureAdClientSecret: string;
+  azureAdTenantId: string;
+};
+type GetAuthOptions = (params: GetAuthOptionsParams) => NextAuthOptions;
+
+export const getAuthOptions: GetAuthOptions = (opts) => ({
+  adapter: PrismaAdapter(prisma),
+  providers: [
+    GoogleProvider({
+      clientId: opts.googleClientId,
+      clientSecret: opts.googleClientSecret,
+    }),
+    AzureAdProvider({
+      clientId: opts.azureAdClientId,
+      clientSecret: opts.azureAdClientSecret,
+      tenantId: opts.azureAdTenantId,
+    }),
+  ],
+  secret: opts.secret,
+  session: { strategy: "jwt" },
+  callbacks: {
+    // JWT is called first, whatever we return is passed to the session callback
+    async jwt({ token, user, account }) {
+      // Initial sign in
+      if (account && user) {
```

Comment:
Is there an advantage of using `account`& `user` over `isNewUser`? Wondering that only because the latter seems more explicit. 

But also fine w/ account & user if we can reference the relevant part of the docs in a comment to make that even easier to follow.

> The arguments user, account, profile and isNewUser are only passed the first time this callback is called on a new session, after the user signs in. In subsequent calls, only token will be available.

[docs](https://next-auth.js.org/configuration/callbacks#:~:text=The%20arguments%20user%2C%20account%2C%20profile%20and%20isNewUser%20are%20only%20passed%20the%20first%20time%20this%20callback%20is%20called%20on%20a%20new%20session%2C%20after%20the%20user%20signs%20in.%20In%20subsequent%20calls%2C%20only%20token%20will%20be%20available.)